diff --git a/lib/core.js b/lib/core.js
index 96e685e7c3a51dd4e8e18b4e869576f95131017a..4af8615107833589cd0496ab962cfe7eb40f881f 100644
--- a/lib/core.js
+++ b/lib/core.js
@@ -237,6 +237,7 @@ const unpluginFactory = (userOptions = {}, metaOptions) => {
   async function runBabelTransform(inputCode, filename, callerName) {
     const result = await (0, _core.transformAsync)(inputCode, {
       babelrc: false,
+      configFile: false,
       filename,
       presets,
       plugins: [...plugins, /\.jsx?/.test(_nodePath.default.extname(filename)) ? _pluginSyntaxFlow.default : [_pluginSyntaxTypescript.default, {
diff --git a/lib/es/core.mjs b/lib/es/core.mjs
index ab666ee586474155be07c94d6839f30fbf7bc194..58473eb2f4e4970d44eabb1bebf5e1ae02e24826 100644
--- a/lib/es/core.mjs
+++ b/lib/es/core.mjs
@@ -228,6 +228,7 @@ export const unpluginFactory = (userOptions = {}, metaOptions) => {
   async function runBabelTransform(inputCode, filename, callerName) {
     const result = await transformAsync(inputCode, {
       babelrc: false,
+      configFile: false,
       filename,
       presets,
       plugins: [...plugins, /\.jsx?/.test(path.extname(filename)) ? flowSyntaxPlugin : [typescriptSyntaxPlugin, {
diff --git a/lib/es/vite.mjs b/lib/es/vite.mjs
index 0da6294e82f05d5a858f8b3740ecdfd8b2dd239c..ab753587da7188d4fa8bae753e40c49064321bf4 100644
--- a/lib/es/vite.mjs
+++ b/lib/es/vite.mjs
@@ -68,6 +68,7 @@ function attachViteHooks(plugin) {
             } catch {}
           }
         }, 150);
+        interval.unref();
         server.httpServer?.once('close', () => clearInterval(interval));
       }
     },
diff --git a/lib/vite.js b/lib/vite.js
index bf3df4c8b81574dafae3f626767706455f03854d..e1be0da27be67d44f3da0200f34413b648538ea2 100644
--- a/lib/vite.js
+++ b/lib/vite.js
@@ -75,6 +75,7 @@ function attachViteHooks(plugin) {
             } catch {}
           }
         }, 150);
+        interval.unref();
         server.httpServer?.once('close', () => clearInterval(interval));
       }
     },
